File: lib/live_view_native_swift_ui/modifiers/drawing_and_graphics/blend_mode.ex
defmodule LiveViewNativeSwiftUi.Modifiers.BlendMode do
  use LiveViewNativePlatform.Modifier

  modifier_schema "blend_mode" do
    field(:blend_mode, Ecto.Enum, values: ~w(
      normal
      darken
      multiply
      color_burn
      plus_darker
      lighten
      screen
      color_dodge
      plus_lighter
      overlay
      soft_light
      hard_light
      difference
      exclusion
      hue
      saturation
      color
      luminosity
      source_atop
      destination_over
      destination_out
    )a)
  end

  def params(blend_mode) when is_atom(blend_mode) and not is_boolean(blend_mode) and not is_nil(blend_mode),
    do: [blend_mode: blend_mode]
  def params(params), do: params
end

File: Sources/LightpandaRenderer/Modifiers/SyntaxConvertible/BlendMode.swift
import SwiftUI
import SwiftSyntax

extension BlendMode: SyntaxConvertible {
    public init?(syntax: some SyntaxProtocol) {
        guard let memberAccess = syntax.as(MemberAccessExprSyntax.self) else {
            return nil
        }
        
        let name = memberAccess.declName.baseName.text
        
        switch name {
        case "normal": self = .normal
        case "multiply": self = .multiply
        case "screen": self = .screen
        case "overlay": self = .overlay
        case "darken": self = .darken
        case "lighten": self = .lighten
        case "colorDodge": self = .colorDodge
        case "colorBurn": self = .colorBurn
        case "softLight": self = .softLight
        case "hardLight": self = .hardLight
        case "difference": self = .difference
        case "exclusion": self = .exclusion
        case "hue": self = .hue
        case "saturation": self = .saturation
        case "color": self = .color
        case "luminosity": self = .luminosity
        case "sourceAtop": self = .sourceAtop
        case "destinationOver": self = .destinationOver
        case "destinationOut": self = .destinationOut
        case "plusDarker": self = .plusDarker
        case "plusLighter": self = .plusLighter
        default: return nil
        }
    }
}
